File: cltch-ntwrk/native-android-kotlin/app/src/main/java/com/cltch/ntwrk/nativeapp/model/Booking.kt
package com.cltch.ntwrk.nativeapp.model

data class Booking(
    val id: String,
    val title: String,
    val venue: String,
    val location: String,
    val dateLabel: String,
    val payLabel: String,
    val status: String,
    val matchScore: Int,
    val city: String,
    val category: String,
    val summary: String,
    val tags: List<String>,
    val timeline: List<BookingTimelineStep>,
    val nextAction: String,
    val counterpartName: String
)

data class BookingTimelineStep(
    val id: String,
    val title: String,
    val detail: String,
    val isComplete: Boolean
)

File: cltch-ntwrk/native-ios-swift/CLTCHNTWRKSwift/Models/Booking.swift
import Foundation

struct Booking: Identifiable, Codable {
    enum State: String, Codable {
        case open
        case pending
        case accepted
        case completed
        case cancelled
    }

    let id: String
    let title: String
    let venue: String
    let location: String
    let dateLabel: String
    let payLabel: String
    let state: State
    let matchScore: Int
    let city: String
    let category: String
    let summary: String
    let tags: [String]
    let timeline: [BookingTimelineStep]
    let nextAction: String
    let counterpartName: String
}

struct BookingTimelineStep: Identifiable, Codable {
    let id: String
    let title: String
    let detail: String
    let isComplete: Bool
}
